File: array/set_based/intersection.py
A = [1, 2, 3, 4, 7];
B = [2, 3, 5, 6, 7, 8];

def union(arr1, arr2):
    asSet = set(arr1);
    reSet = set();
    res = [];

    for i,value in enumerate(arr2):
        if value in asSet:
            res.append(value)

    return res



print(union(A, B));

File: array/set_based/intersection.js
const A = [1, 2, 3, 4, 7];
const B = [2, 3, 5, 6, 7, 8];

function union(arr1, arr2) {
    const asSet = new Set(arr1);
    const reSet = new Set();
    const res = [];

    for (let i = 0; i < arr2.length; i++) {
        if (asSet.has(arr2[i]) && !reSet.has(arr2[i])) {

            reSet.add(arr2[i])
            res.push(arr2[i])
        }
    }

    return res;
}


console.log(union(A, B));